### PR TITLE
Refactor StateBadge component

### PR DESCRIPTION
**Pull request by rkpasia**
| Question         | Response    |
| ---------------- | ----------- |
| Version?         | v1.4.1      |
| Devices tested  | iPhone 6 |
| Related ticket?  | #532        |

---

## Screenshots

| Before   | After    |
| -------- | -------- |
|before|after|
|before|after|

## Description

This PR is related with the update of the StateBadge component. Now it uses styled-components as a driver to style the user interface.

I kept the possibility to override the styles of the StateBadge through the `style` prop. The object that it accept it's a StyleSheet object.

Also I removed from issue-description component the object `styles`. It was only related to the StateBadge component but the styles declared didn't influenced the UI placement of the latter.

### CODE REVIEW

File: src/components/state-badge.component.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { Text, View, StyleSheet } from 'react-native';
+import styled from 'styled-components/native';
```
Comment (machour) [inline src/components/state-badge.component.js]: no need to use '/native' anymore, just use this:

```
import styled from 'styled-components';
```

File: src/components/state-badge.component.js [hunk 2]
```diff
@@ -58,19 +47,25 @@ export const StateBadge = ({
...
+      issueStateColor = colors.purple;
+      break;
+    case 'open':
+      issueStateColor = colors.green;
+      break;
+    case 'closed':
+      issueStateColor = colors.red;
+      break;
+    default:
+      issueStateColor = colors.gray;
```
Comment (machour) [inline src/components/state-badge.component.js]: could you do something like this instead for a better readability?

```js
const stateColor = {
  merged: colors.purple,
  open: colors.green,
  closed: colors.red,
};
const issueStateColor = stateColor[issueState] ? stateColor[issueState] : colors.gray;
```

File: src/components/issue-description.component.js [hunk 3]
```diff
@@ -15,13 +15,6 @@ import { translate, relativeTimeToNow } from 'utils';
 import { colors, fonts, normalize } from 'config';
 import { v3 } from 'api';
 
-const styles = StyleSheet.create({
```
Comment (machour) [inline src/components/issue-description.component.js]: why was this removed?

File: src/components/state-badge.component.js [hunk 4]
```diff
@@ -58,19 +47,25 @@ export const StateBadge = ({
...
   }
 
   return (
-    <View style={[styles.badge, style, issueStyle]}>
-      <Text style={styles.text}>{issueText}</Text>
-    </View>
+    <Badge color={issueStateColor} style={style}>
```
Comment (machour) [inline src/components/state-badge.component.js]: You are not passing `style` anymore from `issue-description.component.js`, but you're still using the prop here .. Could you give it a second look?

Comment (rkpasia) [reply src/components/state-badge.component.js]: Yeah I thought to get completely rid of the style prop but it could still be useful in case there is a need to override something from a parent component without modifying again this one instead.

But it's up for discussion, don't know if it is really needed or not so I left it there 😆

File: src/components/issue-description.component.js (hunk 3 above)
Comment (rkpasia) [reply src/components/issue-description.component.js]: It was used just for the Badge styles so I removed it. I don't think is needed here, also styled-components are already imported.
Also the styles defined for the Badge don't change the UI so they are useless for what I can saw.

**Comment by rkpasia** [comment]
Don't really know how the tests work yet 😆 I'll get through them tomorrow 👍

**Comment by machour** [comment]
No worries mate! We use jest for testing, and everything is located under `<root>/__tests__`
You can run the full test suite in your console like this:

```
yarn test
```

or you can run a single test like this (which is waaaayyyy faster when fixing something 😄) :
```
yarn test __tests__/tests/components/StateBadge.js
``` 

happy debugging 🕺

**Comment by rkpasia** [comment]
What should I do to augment the coverage?

File: src/components/state-badge.component.js (hunk 4 above)
Comment (chinesedfan) [reply src/components/state-badge.component.js]: Prop `style` is also used in `repository-section-title.component.js`, which is a useless component. I remembered @housseindjirdeh confirmed that in somewhere, but I forgot the detail issue/pr number. I will collect feedbacks in #725.

Comment (machour) [reply src/components/state-badge.component.js]: it was on Gitter, I added the relevant screenshot in the issue you opened.
